File: taskcenter/src/main/java/com/iwdael/taskcenter/core/TaskRunnable.java
package com.iwdael.taskcenter.core;

import com.iwdael.taskcenter.TaskCenter;

/**
 * @author : iwdael
 * @mail : iwdael@outlook.com
 * @project : https://github.com/iwdael/TaskCenter
 */
public class TaskRunnable implements Runnable {
    private final Object task;
    private final String taskId;
    private final TaskProcessor taskProcessor;
    private final TaskCenter.Config config;

    public TaskRunnable(TaskProcessor taskProcessor, TaskCenter.Config config, Node<Object, Object, Object> node, Object task, String taskId) {
        this.taskProcessor = taskProcessor;
        this.config = config;
        this.task = task;
        this.taskId = taskId;
        this.taskProcessor.initPersistence();
        this.taskProcessor.stagePrepared(node, task);
    }

    @Override
    public void run() {
        try {
            taskProcessor.dispatch();
        } catch (Exception exception) {
            if (exception instanceof InterruptedException) {
                taskProcessor.notifySuspend();
                throw exception;
            } else {
                exception.printStackTrace();
                taskProcessor.notifyError();
            }
        }
    }
}

File: taskcenter/src/main/java/com/iwdael/taskcenter/core/TaskRunnable.kt
package com.iwdael.taskcenter.core

import com.iwdael.taskcenter.TaskCenter
import com.iwdael.taskcenter.util.Logger

class TaskRunnable(private val source: Any) : Runnable {
    companion object {
        private const val TAG = "TaskLauncher"
    }

    override fun run() {
        val node: Node<Any, Any>? = TaskCenter.node(source::class.java)
        if (node == null) {
            Logger.tag(TAG).e("not support class:${source::class.java.name}")
            return
        }
        val taskProcessingCenter = TaskProcessingCenter(node)
        taskProcessingCenter.default(node, source)
        taskProcessingCenter.handle()
    }


}